testplan/exporters/testing/pdf/__init__.py

```diff
@@ -213,18 +220,32 @@ class PDFExporter(Exporter):
 
     CONFIG = PDFExporterConfig
 
-    def export(self, source):
-
-        pdf_path = self.cfg.pdf_path
+    def __init__(self, **options):
+        super(PDFExporter, self).__init__(**options)
+        self.name = "PDF exporter"
 
+    def export(self, source):
         if len(source):
-            create_pdf(source, self.cfg)
-            self.logger.exporter_info(
-                "PDF generated at %s", os.path.abspath(pdf_path)
-            )
-            self.url = "file:{}".format(
-                pathname2url(os.path.abspath(pdf_path))
-            )
+            try:
+                if self.cfg.interactive_port is None:
+                    override = True
+                else:
+                    override = False
+            except AttributeError:
+                override = True
+            pdf_path = create_pdf(source, self.cfg, override=override)
```

Comment:
I suggest that we move the check of cfg.interactive_port to create_pdf(), so that we can save the override parameter and there we are checking if the file already exists anyway.